### DISCUSSION TITLE
Machine learning implementation with flet

### DISCUSSION THREAD

**Discussion by WaniAthar** [flet-dev/flet#1829] [Ideas]
**Please Describe The Problem To Be Solved**

Flet is a lightweight UI kit in Python it should have support for the machine learning models like object detection, image classification, or any other model, this will be very useful as machine learning models are generally written in Python and flet is also written in python, So there should be a way to integrate both.

**(Optional): Suggest A Solution**

(Replace This Text: A concise description of your preferred solution. Things to address include:
* Details of the technical implementation, e.g. Flutter widget solving the problem.
* Tradeoffs made in design decisions
* Caveats and considerations for the future

If there are multiple solutions, please present each one separately. Save comparisons for the very end.)

**Comment by Skquark** [comment]
You can use any of the existing Python libraries to integrate with the Flet UI without too much difficulty.  I've already implemented a pretty complete AI Toolkit with Flet that works well, but the main limitation is getting the real-time updates on images since it needs to save as an image file first then apply to the Image widget rather than direct PIL or Numpy.  I wasn't as interested in the Object Detection, but have found python projects to do it.
If you want to play around with my machine learning Flet project, check out https://StableDiffusionDeluxe.com where you can generate images, videos, audio AI, model training and conversion, some image classification, and a bunch of other extras.... Might not be specifically what you're asking to create, but it's flexible and may put you on the right path.  I'm always looking for new opensource python AI features to integrate in there..

**Comment by WaniAthar** [comment]
> You can use any of the existing Python libraries to integrate with the Flet UI without too much difficulty. I've already implemented a pretty complete AI Toolkit with Flet that works well, but the main limitation is getting the real-time updates on images since it needs to save as an image file first then apply to the Image widget rather than direct PIL or Numpy. I wasn't as interested in the Object Detection, but have found python projects to do it. If you want to play around with my machine learning Flet project, check out https://StableDiffusionDeluxe.com where you can generate images, videos, audio AI, model training and conversion, some image classification, and a bunch of other extras.... Might not be specifically what you're asking to create, but it's flexible and may put you on the right path. I'm always looking for new opensource python AI features to integrate in there..

Thanks for your reply, but actually i am developing desktop app in which i want to detect objects in real time and based on the detections some output has to come.

**Comment by monkeycc** [comment]
I am also developing a desktop client
I also encountered the same problem

we can read Image with OpenCV. We can view frames in flet control.
https://flet.dev/docs/controls/image/Image

You'll find both example for camera and video source reading in the following repo.
https://github.com/ipritom/flet-demo/tree/main/media

I like this yolo framework
https://github.com/ultralytics/ultralytics


I am also developing a desktop client
I also encountered the same problem
https://github.com/flet-dev/flet/discussions/1162

**Comment by goyalyashpal** [comment]
* also see this: https://github.com/flet-dev/flet/discussions/1209

**Comment by Vaibhavsun** [comment]
Actually is there any way to create machine learning android apps using flet as it found very difficult to integrate opencv and tensorflow libraries for this
